pkg/pki/ca/ca.go

```diff
@@ -56,24 +74,57 @@ type IstioCA struct {
 }
 
 // NewSelfSignedIstioCA returns a new IstioCA instance using self-signed certificate.
-func NewSelfSignedIstioCA(caCertTTL, certTTL time.Duration, org string) (*IstioCA, error) {
-	now := time.Now()
-	options := CertOptions{
-		NotBefore:    now,
-		NotAfter:     now.Add(caCertTTL),
-		Org:          org,
-		IsCA:         true,
-		IsSelfSigned: true,
-		RSAKeySize:   caKeySize,
-	}
-	pemCert, pemKey := GenCert(options)
+func NewSelfSignedIstioCA(caCertTTL, certTTL time.Duration, org string, namespace string,
+	core corev1.CoreV1Interface) (*IstioCA, error) {
 
+	caSecret, err := core.Secrets(namespace).Get(CASecret, metav1.GetOptions{})
```

Comment:
I feel it deserves some detailed comment about what is happening here.

Reply:
done